New package: OffsetTables v0.0.1

JuliaRegistrator opened the pull request:
- Registering package: OffsetTables
- Repository: https://github.com/LilithHafner/OffsetTables.jl
- Created by: @LilithHafner
- Version: v0.0.1
- Commit: 5ccc765b5df9b3501774d859a6a72f0acf274224
- Reviewed by: @LilithHafner
- Reference: https://github.com/LilithHafner/OffsetTables.jl/commit/5ccc765b5df9b3501774d859a6a72f0acf274224#commitcomment-140212218
- Description: An efficient sampler for discrete random variables
<!-- bf0c69308befbd3ccf2cc956ac8a46712550b79fc9bfb5e4edf8f833f05f4c18b06eddad8845b45beb9f45c2b8020dd636a0414f037d2d4b8c2c6f81a88681491004e5b8c3f203ac2b245e57d23a1a51679721be674bddcb9d6660beeafeda2095568c8a3d0affa1dc0c3c7bfe3ea41f028799cf821b55b6ec1abb536a3d61bcdb0ed249da695160f0ffa59f3e18fff51e7d98d375c4cbef1d646c53a1e99c520c7fdfc951ec2ae1b4a7be5c26575145ba5bafbf8478774f3b3882f0eb0f7441dcb8cf5b8c7cedfa590721454312cfc9da5496290bbe86b66f37c21af42632ad -->

goerz commented:
Can you extend the README a bit with a paragraph to give context for the package and maybe a small usage example? As triage members, we should probably lead by example when it comes to (not) registering "undocumented" packages, no? ;-)

LilithHafner commented:
The package was pretty solid from functionality, tests, efficiency & benchmarks perspectives. But you're right that that does not make up for missing docs.

I can and should do better when it comes to documentation quality.

The [new docs](https://offsettables.lilithhafner.com) are still a work in progress, but they are reasonable.

goerz commented:
[noblock] Thanks! Looks great!

I don't mind "early stage" docs in registrations, but I'm pretty strict about the README having a paragraph that explains the purpose of the package, and then either a small usage example or a link to a more complete online documentation.

P.S.: don't forget to add `[noblock]` to your own comment

LilithHafner commented:
A note on the CI failure: This package requires Julia 1.11 because it uses `Memory`.

A note on the name: This package implements [Alias tables](https://en.wikipedia.org/wiki/Alias_method) using diff coding to store the aliases as offsets rather than absolute aliases. This diff coding saves about one instruction in sampling and is an implementation detail.

I'm open to other suggestions for the name (e.g. AliasTables.jl)

JuliaRegistrator committed:
New package: OffsetTables v0.0.1

UUID: 66dad0bd-aa9a-41b7-9441-69ab47430ed8
Repo: https://github.com/LilithHafner/OffsetTables.jl.git
Tree: 25af3f7f4143d068403337783e6079a395ad3ad8

Registrator tree SHA: 17aec322677d9b81cdd6b9b9236b09a3f1374c6a

lgoettgens commented:
> A note on the name: This package implements [Alias tables](https://en.wikipedia.org/wiki/Alias_method) using diff coding to store the aliases as offsets rather than absolute aliases. This diff coding saves about one instruction in sampling and is an implementation detail.
> 
> I'm open to other suggestions for the name (e.g. AliasTables.jl)

There might be possible confusion with https://github.com/JuliaArrays/OffsetArrays.jl. I have no opinion on this but wanted to make you aware of this.

goerz commented:
[noblock]

> This package implements [Alias tables](https://en.wikipedia.org/wiki/Alias_method) using diff coding to store the aliases as offsets rather than absolute aliases. This diff coding saves about one instruction in sampling and is an implementation detail.

Based on that description, it does seem like `AliasTables` would be a better name. But this is pretty far outside my domain of expertise, so I'm not sure that I have much insight here.

Even with the current name, though, I don't really see much potential for confusion with `OffsetArrays`. @lgoettgens: Do you feel strongly enough about the potential similarity to make your comment blocking, or did you just forget a `[noblock]`?

LilithHafner commented:
Renamed to AliasTables.jl. New PR: https://github.com/JuliaRegistries/General/pull/103755
